docs/recipes/stencil.md

```diff
@@ -0,0 +1,85 @@
+# Usage with Stencil
+
+[Stencil](https://stenciljs.com/) web components work very well with XState.
+
+### components/toggle/toggle-machine.ts
+
+```js
+import { Machine } from "xstate";
+
+// this machine is completely decoupled from stencil
+export const toggleMachine = Machine({
+  id: "toggle",
+  initial: "inactive",
+  states: {
+    inactive: {
+      on: { toggle: "active" }
+    },
+    active: {
+      on: { toggle: "inactive" }
+    }
+  }
+});
+```
+
+### components/toggle/toggle.tsx
+
+Add a `state` property to your component, decorated with `@State` so that it triggers a re-render when changed.
+
+On `componentWillLoad`, interpret the `toggleMachine` and listen for state transitions.
+
+After a transition has occured, the `state` property is set to the machine's new state, triggering a re-render.
+
+```js
+import { Component, h, State, Prop } from "@stencil/core";
+import { interpret } from "xstate";
+import { toggleMachine } from "./toggle-machine";
+
+@Component({
+  tag: "my-toggle",
+  styleUrl: "toggle.css",
+  shadow: true
+})
+export class Toggle {
+  @State() state;
+
+  private _service;
+
+  componentWillLoad() {
+    this._service = interpret(toggleMachine).onTransition(current => {
+      this.state = { current };
+    });
+
+    this._service.start();
+  }
+
+  componentDidUnload() {
+    this._service.stop();
+  }
+
+  render() {
+    const { current } = this.state;
+    const { send } = this._service;
+
+    return (
+      <button onClick={() => send("toggle")}>

```

Comment:
You're not defining `send` anywhere?

Reply:
it's being destructured from `this._service`

Reply:
🤦🏻‍♂️ I'm blind.

BTW, I'd do the `send` like this:

```ts
export class Toggle {
  // ...

  toggle = () => this._service.send("toggle");

  render() {
    return (
      <button onClick={this.toggle}>...</button>
    );
  }
}
```

Because that way `onClick` always binds to the same reference, whereas binding to a new function reference (anonymous function) on each render would be detected as a prop change and re-render the element. Micro-optimization though... so anyway, whatever you find more clear.